opensearch/services/datatypes/datatypes.php

```diff
@@ -0,0 +1,430 @@
+	* @return string query header component (e.g. 'SELECT ?id ?label')
+	*/
+	public function getQueryHeader($type) {
+	
+		$header = "";
+		switch($type) {
+			case "datatypes":
+				$header .= "?datatype ?dco_id ?label ?year ?parameterLabel ?access ";
+				$header .= '(GROUP_CONCAT(DISTINCT ?creator ; SEPARATOR=",") AS ?creator) ';
+				$header .= '(GROUP_CONCAT(DISTINCT ?creator_label ; SEPARATOR=",") AS ?creator_label) ';
+				$header .= '(GROUP_CONCAT(DISTINCT ?sourceStandard ; SEPARATOR=",") AS ?sourceStandard) ';
+                $header .= '(GROUP_CONCAT(DISTINCT ?sourceStandard_label ; SEPARATOR=",") AS ?sourceStandard_label) ';
+				break;
+			case "count":
+				$header .= "(count(DISTINCT ?datatype) AS ?count)";
+				break;
+			default:
+				$header .= "?id ?label (COUNT(DISTINCT ?datatype) AS ?count)";
+				break;
+		}
+		return $header;
+	}
+	
+	/**
+	* Return SPARQL query footer component
+	* @param string $type search type (e.g. 'parameters', 'authors', 'keywords')
+	* @param int $limit size of result set
+	* @param int $offset offset into result set
+	* @param string $sort query result sort parameter
+	* @return string query footer component (e.g. 'GROUP BY ?label ?id')
+	*/
+	public function getQueryFooter($type, $limit=null, $offset=0, $sort=null) {
+	
+		$footer = "";
+		switch($type) {
+			case "datatypes":
+				$footer .= " GROUP BY ?datatype ?dco_id ?label ?year ?parameterLabel ?access";
+				$footer .= " ORDER BY ?label";
+				if ($limit)	$footer .= " LIMIT $limit OFFSET $offset";
+				break;
+			case "count":
+				break;
+			default:
+				$footer .= " GROUP BY ?label ?id";
+				break;
+		}
+		return $footer;
+	}
+	
+	/**
+	  * Return SPARQL query WHERE clause minus constraint clauses for specified search type
+	  * @param string $type search type (e.g. 'datasets', 'authors', 'keywords')
+	  * @return string WHERE clause component minus constraint clauses (e.g. '?dataset a dcat:Dataset . ')
+	  */
+	public function getQueryBody($type) {
+		
+		$body = "";
+		switch($type) {
+			case "creationYear":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype  dco:createdAtTime ?t . ";
+				$body .= "BIND(str(?t) AS ?year) . ";
+				break;
+				
+			case "creator":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype prov:qualifiedAttribution [prov:hadRole ?r ] . ";
+				$body .= "?r a prov:Role . ";
+				$body .= "?c rdf:type ?r . ";
+				$body .= "?c a dco:creator . ";
+				$body .= "BIND(str(?c) AS ?creator) . ";
+				break;
+
+			case "parameter":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype dco:hasParameter [rdfs:label ?l] . ";
+				$body .= "BIND(str(?l) AS ?parameterLabel) . ";
+				break;
+
+			case "sourceStandard":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype dco:sourceStandard [rdfs:label ?l] . ";
+				$body .= "BIND(str(?l) AS ?sourceStandard_label) . ";
+				break;
+
+			case "subjectArea":
```

Comment:
Should it be like this? I checked the SPARQL query using this and the results were coming.
				$body .= "?datatype a dco:DataType . ";
				$body .= "?datatype dco:dataTypeSubjectArea ?id . ";
				$body .= "?id a skos:Concept . ";
				$body .= "?id rdfs:label ?l .";
				$body .= "BIND(str(?l) AS ?label) . ";